R/compare_diagnoses.R

```diff
@@ -0,0 +1,278 @@
+                                   df2= bootstrap_df2,
+                                   comparison = comparison_list,
+                                   diagnosands_list = diagnosands) {
+    # Estimated diagnosands and se(diagnosands) from each design
+    d1 <- df1[[label]]
+    d2 <- df2[[label]]
+    comp <- comparison[[label]]
+    c_names <- colnames(comp)
+    set <-   grepl("^estimand_label", c_names)| grepl("^estimator_label", c_names)| grepl("^term", c_names) 
+    select_cols <- c_names[grepl("^design_label", c_names)]
+    select_cols <- c(select_cols, c_names[set])
+    identifiers <- comp[,   select_cols]
+    
+    # Compute difference stats
+    difference <- do.call(rbind, lapply(diagnosands_list, function(diagnosand){
+      diagnosand1 <- comp[,paste0(diagnosand, "_1")]
+      diagnosand2 <- comp[,paste0(diagnosand, "_2")]
+      se1 <- comp[,paste0("se(",diagnosand, ")_1")]
+      se2 <- comp[,paste0("se(",diagnosand, ")_2")]
+      diff <- d2[, diagnosand] - d1[, diagnosand] 
+      mu <- mean(diff) 
+      p.value <- ifelse(sign(mu) == -1, mean(diff >= 0),  mean(diff <= 0))
+      
+      # Combine
+     out<- data.frame(diagnosand =diagnosand,
+                 mean_1 = diagnosand1,
+                 mean_2 = diagnosand2,
+                 mean_difference = mu ,
+                 se_1 = se1,
+                 se_2 = se2,
+                 se_difference = sd(diff),
+                 p.value = p.value,
+                 sims = sims, 
+                 bootstrap_sims = bootstrap_sims,
+                stringsAsFactors = FALSE)
+     
+      }))
+    suppressWarnings(cbind(identifiers, difference))
+
+  }
+  
+  return_df <- lapply(names(comparison_list),  create_difference_df )
+  return_df <- data.frame(rbind_disjoint(return_df, infill = ""))
+
+  
+  return_list <- list(compared_diagnoses_df =  return_df, 
+                      diagnosis1 = diagnosis1, 
+                      diagnosis2 = diagnosis2)
+  
+  class(return_list) <- "compared_diagnoses"
+  
+  return_list
+  
+}
+
+
+#' @export
+print.compared_diagnoses <- function(x, ...) {
+  print(summary(x))
+  invisible(x)
+}
+
+
+#' @export
+summary.compared_diagnoses <- function(object, ...) {
+  structure(object, class = c("summary.compared_diagnoses", "data.frame"))
+  
+}
+
+#' @export
+print.summary.compared_diagnoses <- function(x, ...){
+  
+  bootstrap_sims <- x$diagnosis1$bootstrap_sims
+  sims <- nrow(x$diagnosis1$simulations_df)
+  compared_diagnoses_df <- x[["compared_diagnoses_df"]]
+
+  column_names <- colnames(compared_diagnoses_df)
+  identifiers <- column_names %in% c("estimand_label", "term") | grepl("^estimator_label", column_names)
+  identifiers_columns <- column_names[identifiers]
+  p.value <- compared_diagnoses_df$p.value
+  stars <- ifelse(p.value < 0.01 ,"***",
+                  ifelse(p.value < 0.05 & p.value>=0.01, "**",
+                         ifelse(p.value < 0.1 & p.value>=0.05, "*", "")))
```

Comment:
We should match the stars to base R / `stats::lm` - a p of .08 is not a `*`

See comment in discussion thread.